provider/cluster/service.go

```diff
@@ -109,22 +123,101 @@ loop:
 			case event.ManifestReceived:
 				s.log.Info("manifest received")
 
-				// TODO: determine group to deploy.
-				if len(ev.Manifest.Groups) != 1 {
-					s.log.Error("indeterminate group")
+				mgroup := ev.ManifestGroup()
+				if mgroup == nil {
+					s.log.Error("indeterminate manifest group", "lease", ev.LeaseID, "group-name", ev.Group.Name)
 					break
 				}
 
-				if err := s.client.Deploy(ev.LeaseID.OrderID(), ev.Manifest.Groups[0]); err != nil {
-					s.log.Error("deploying", "err", err)
+				key := ev.LeaseID.OrderID().String()
+
+				state := s.deployments[key]
+
+				if state == nil {
+					s.log.Error("lease received without reservation")
+					break
 				}
 
-				s.log.Info("deploy done")
+				if state.monitor == nil {
+					state.monitor = newDeploymentMonitor(s, ev.LeaseID, ev.Group, mgroup)
+					break
+				}
+
+				if err := state.monitor.update(mgroup); err != nil {
+					s.log.Error("updating deployment", "err", err)
+				}
+
+			case *event.TxCloseDeployment:
+
+				// teardown/undeploy managed deployments
+
+				for _, dm := range s.deployments {
+					if bytes.Compare(ev.Deployment, dm.reservation.OrderID().DeploymentID()) == 0 {
```

Comment:
should use bytes.Equal(ev.Deployment, dm.reservation.OrderID().DeploymentID()) instead